src/rgw/rgw_req_context.h

```diff
@@ -0,0 +1,24 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "common/async/yield_context.h"
+namespace opentelemetry::trace {
+  class Span;
+}
+
+class DoutPrefixProvider;
+
+// this struct holds information which is created at the frontend
+// and should trickle down through all function calls to the backend
+struct req_context {
+  const DoutPrefixProvider* const dpp{nullptr};
+  const optional_yield y{null_yield};
+  opentelemetry::trace::Span* const span{nullptr};
```

Comment:
given a raw pointer to `Span`, it doesn't look like our `src/common/tracer.h` interfaces would let us add a child span, because `Tracer::add_span()` takes the parent span by shared_ptr

Reply:
made it a raw pointer again.
no need for a new `Tracer::add_span()` API, as I'm just using the `SpanContext` API.
see: https://github.com/ceph/ceph/pull/53547/files#diff-d07015a33dfd523c87c32b6eef5fb7b564abe0e648875c7fea21922d21565020R496